app/services/search/postgres/listing.rb

```diff
@@ -0,0 +1,58 @@
+module Search
+  module Postgres
+    class Listing
+      ATTRIBUTES = %i[
+        id
+        body_markdown
+        bumped_at
+        cached_tag_list
+        classified_listing_category_id
+        contact_via_connect
+        expires_at
+        organization_id
+        originally_published_at
+        location
+        processed_html
+        published
+        slug
+        title
+        user_id
+      ].freeze
+
+      DEFAULT_PER_PAGE = 75
+      MAX_PER_PAGE = 150 # to avoid querying too many items, we set a maximum amount for a page
+
+      def self.search_documents(category: nil, page: 0, per_page: DEFAULT_PER_PAGE, term: nil)
+        # NOTE: [@rhymes/atsmith813] we should eventually update the frontend
+        # to start from page 1
+        page = page.to_i + 1
+        per_page = [(per_page || DEFAULT_PER_PAGE).to_i, MAX_PER_PAGE].min
+
+        relation = ::Listing.where(published: true)
```

Comment:
I think we should preload users and categories, otherwise we get a N + 1 of SQL calls when `.author` or `.category` are serialized